mesonbuild/build.py

```diff
@@ -2114,6 +2114,11 @@ def post_init(self) -> None:
             if self.rust_crate_type == 'staticlib':
                 # FIXME: In the case of no-std we should not add those libraries,
                 # but we have no way to know currently.
+
+                # XXX:
+                #  In the case of no-std, we are likely in a bare metal case
+                #  and thus, machine_info kernel should be set to 'none'.
+                #  In that case, native_static_libs list is empty.
```

Comment:
These two comments seem to be in conflict now.

Reply:
Indeed, this is more a reply to the @xclaesse comment above.
I am not legitimate to remove `FIXME` from maintainers (IMHO), if you all agree with the proposed fixe, i could squash/rephrase those comments.

Does it need a note in `Rust` chapter in documentation ?
As i said in PR description (and commit log), building no-std requires to use `--target` rustc option and thus overriding rust in ths cross file.
 > rust = ['rustc', '--target', '<rustc valid target>']